src/lib/hooks/useScrollPositionHook.js

```diff
@@ -0,0 +1,31 @@
+import { useLayoutEffect } from 'react';
+import { getElementsPositionDifference } from '../services/elementPositionService';
+
+export const useScrollPosition = (effect, dependencies, contentEl, viewportEl, wait) => {
+  let throttleTimeout = null;
+
+  const callBack = () => {
+    effect(getElementsPositionDifference(contentEl, viewportEl));
+    throttleTimeout = null;
```

Comment:
I am wondering, what's the reason to have throttleTimeout here? When we do not use wait than throttleTimeout = null is not necessary here, isn't it?

Reply:
Resolved, see above.